.github/workflows/release.yml

```
@@ -58,3 +63,56 @@ jobs:
           message: |
             :ghost: [${{ github.repository }}] Release *${{ github.ref_name }}* didn't get triggered in Buildkite.
             Build: (<${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|here>)
+
+  post-release:
+    name: "Bump versions and create PR"
+    runs-on: ubuntu-latest
+    needs:
+      - release
+    permissions:
+      contents: write
+    steps:
+      - uses: elastic/apm-pipeline-library/.github/actions/github-token@current
+        with:
+          url: ${{ secrets.VAULT_ADDR }}
+          roleId: ${{ secrets.VAULT_ROLE_ID }}
+          secretId: ${{ secrets.VAULT_SECRET_ID }}
+
+      - uses: elastic/apm-pipeline-library/.github/actions/setup-git@current
+        with:
+          username: ${{ env.GIT_USER }}
+          email: ${{ env.GIT_EMAIL }}
+          token: ${{ env.GITHUB_TOKEN }}
+
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref || 'main' }}
+          token: ${{ env.GITHUB_TOKEN }}
+
+      - name: Set up JDK ${{ env.JAVA_VERSION }}
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+          distribution: ${{ env.JAVA_DIST }}
+          cache: 'maven'
+
+      - name: Create the release tag
+        run: |
+          git tag "v${{ env.RELEASE_VERSION }}"
+          git push origin "v${{ env.RELEASE_VERSION }}"
+
+      - name: Create a post release branch
+        run: |
+          git checkout -b "post-release-v${{ env.RELEASE_VERSION }}"
+
+      - name: Bump versions
+        run: |
+          ./mvnw -V versions:set -DprocessAllModules=true -DgenerateBackupPoms=false -nextSnapshot=true
+          git add --all
+          git commit -m "chore: prepare for next iteration"
+
+      - name: Push the post release branch
+        run: git push origin "post-release-v${{ env.RELEASE_VERSION }}"
+
+      - name: Create the post release PR
+        run: gh pr create --title="Post release process" --base main --head "post-release-v${{ env.RELEASE_VERSION }}" -b "Prepare for next iteration"
```

Make the post-release PR name consistent with the prepare PR by adding release version.
```suggestion
        run: gh pr create --title="Post Release v${{ env.RELEASE_VERSION }}" --base main --head "post-release-v${{ env.RELEASE_VERSION }}" -b "Prepare for next iteration"
```

---

Ref: https://github.com/elastic/ecs-logging-java/pull/226/commits/ae99e9780a0e9cae0b7c9fcd8ef67bfb5c2acbb6